wntr/network/model.py

```diff
@@ -1432,6 +1432,140 @@ def reset_initial_values(self):
+            except KeyError: pass
+            node._leak_demand = None
+            node._leak_status = False
+            node._is_isolated = False
+
+        for name, node in self.nodes(Tank):
+            node._head = None
+            node._demand = None
+            node._pressure = None
+            node.init_level = float(results.node['head'].loc[end_time, name] - node.elevation)
+            try: node.initial_quality = float(results.node['quality'].loc[end_time, name])
+            except KeyError: pass
+            node._prev_head = node.head
+            node._leak_demand = None
+            node._leak_status = False
+            node._is_isolated = False
+
+        for name, node in self.nodes(Reservoir):
+            node._head = None
+            node._demand = None
+            node._pressure = None
+            try: node.initial_quality = float(results.node['quality'].loc[end_time, name])
+            except KeyError: pass
+            node._leak_demand = None
+            node._is_isolated = False
+
+        for name, link in self.links(Pipe):
+            link.initial_status = results.link['status'].loc[end_time, name]
+            try: link.initial_setting = results.link['setting'].loc[end_time, name]
+            except KeyError: link.initial_setting = link.setting
+            link._user_status = link.initial_status
+            link._internal_status = LinkStatus.Active
+            link._is_isolated = False
+            link._flow = None
+            link._prev_setting = None
+
+        for name, link in self.links(Pump):
+            link.initial_status = results.link['status'].loc[end_time, name]
+            try: link.initial_setting = results.link['setting'].loc[end_time, name]
+            except KeyError: link.initial_setting = link.setting
+            link._user_status = link.initial_status
+            link._setting = link.initial_setting
+            link._internal_status = LinkStatus.Active
+            link._is_isolated = False
+            link._flow = None
+            if isinstance(link, PowerPump):
+                link.power = link._base_power
+            link._prev_setting = None
+
+        for name, link in self.links(Valve):
+            link.initial_status = results.link['status'].loc[end_time, name]
+            try: link.initial_setting = results.link['setting'].loc[end_time, name]
+            except KeyError: link.initial_setting = link.setting
+            # print(name, link.initial_status, link.initial_setting)
+            link._user_status = link.initial_status
+            link._setting = link.initial_setting
+            link._internal_status = LinkStatus.Active
+            link._is_isolated = False
+            link._flow = None
+            link._prev_setting = None
+
+        to_delete = []
+        for name, control in self.controls():
+            control._reset()
+            still_good = control._shift(end_time)
```

Comment:
We have an issue here with the time shifting. We need to make sure that the controls are only shifted when we change pattern start, not when we set conditions, per se.